Clouseau: Add a few more gimmicks for numbers

scymtym opened the pull request.

kc- commented on Apps/Clouseau/src/objects/number.lisp:
```diff
@@ -48,19 +78,66 @@
+    ;; Length in bits
     (format-place-row stream object 'reader-place 'integer-length
-                      :label "Length")))
+                      :label "Length")
```
Maybe name this Bit-length? Or possibly "Length: 3 (bits)"

kc- commented on Apps/Clouseau/src/objects/number.lisp:
```diff
@@ -48,19 +78,66 @@
+                                  double-float))
+              ,(constant 'pi)))))
+    (limits)))
+
```
A matter of taste, but for integers, the `(type fixnum 0 4611686018427387903)` could maybe benefit from being shortened (or labelled with a badge), like `(type fixnum 0 most-positive-fixnum)`

scymtym commented:
> I noted two suggestions in the commit comments.

Thank you.

I changed the label from "Length" to "Length (bits)" (and added the Hamming weight, while at it).

For the type in the header, I disabled printing it if it is a cons.

kc- commented:
Looks good. Only one more question: for the prime factorization, it feels a little unusual to explicitly denote the exponent for singly-repeated factors. Was this a conscious decision?

scymtym commented:
> Looks good. Only one more question: for the prime factorization, it feels a little unusual to explicitly denote the exponent for singly-repeated factors. Was this a conscious decision?

Just laziness. I will change it.

scymtym commented:
Fixed.

kc- commented:
lgtm, thank you.
